File: src/main/java/me/zoarial/NetworkArbiter/exceptions/DuplicatePlacement.java
package me.zoarial.NetworkArbiter.exceptions;


import me.zoarial.NetworkArbiter.annotations.ZoarialObjectElement;

import java.lang.reflect.Field;

public class DuplicatePlacement extends ArbiterException {
    public DuplicatePlacement(String str) {
        super(str);
    }

    public DuplicatePlacement(ZoarialObjectElement existingElement, ZoarialObjectElement duplicateElement) {
        super("Duplicate placement found: " + existingElement.placement() + ". Existing: \"" + existingElement + "\". New: \"" + duplicateElement + "\".");

    }
}

File: src/main/java/me/zoarial/networkArbiter/exceptions/DuplicatePlacement.kt
package me.zoarial.networkArbiter.exceptions

import me.zoarial.networkArbiter.annotations.ZoarialObjectElement

class DuplicatePlacement : ArbiterException {
    constructor(str: String?) : super(str) {}
    constructor(existingElement: ZoarialObjectElement, duplicateElement: ZoarialObjectElement) : super("Duplicate placement found: " + existingElement.placement + ". Existing: \"" + existingElement + "\". New: \"" + duplicateElement + "\".") {}
}